daemon/delete.go

```diff
@@ -102,8 +102,10 @@ func (daemon *Daemon) cleanupContainer(container *container.Container, forceRemo
 	// Save container state to disk. So that if error happens before
 	// container meta file got removed from disk, then a restart of
 	// docker should not make a dead container alive.
-	if err := container.ToDiskLocking(); err != nil {
-		logrus.Errorf("Error saving dying container to disk: %v", err)
+	if _, err := os.Stat(container.Root); err == nil {
```

Comment:
@shishir-a412ed 

It would return a `no such file or directory` error which we could catch up.

Which make me realize that it is already the error being returned.

So a simple

```go
if err := container.ToDiskLocking(); err != nil && !os.IsNotExist(err) {
    logrus.Errorf("Error saving dying container to disk: %v", err)
}
```

would be sufficient.